app/src/main/java/com/example/diceroller/MainActivity.kt

```diff
@@ -4,122 +4,80 @@ import android.os.Bundle
+        setContentView(binding.root)
         supportActionBar?.hide()
 
-        viewModel = ViewModelProvider(this).get(RolledDiceViewModel::class.java)
+        viewModel = ViewModelProvider(this)[DiceRollerViewModel::class.java]
 
         setListeners()
+        observeState()
     }
 
-    private fun setListeners() {
-        val navigationMenu = binding.navView
-
-        val menuButton = binding.menuButton
-
-        toggleNavigation(menuButton, navigationMenu)
-
-        navigationMenu.setNavigationItemSelectedListener(this)
-    }
-
-
-    private fun toggleNavigation(menuButton: View, navigationMenu: View) {
-        menuButton.setOnClickListener {
-            when (navigationMenu.visibility) {
-                View.GONE -> {
-                    navigationMenu.visibility = View.VISIBLE
-                }
-                View.VISIBLE -> {
-                    navigationMenu.visibility = View.GONE
-                }
-                else -> {
-                    navigationMenu.visibility = View.GONE
+    private fun observeState() {
+        lifecycleScope.launchWhenStarted {
+            viewModel.viewState.collect {
+                if (it is ViewState.RollDie) {
+                    binding.title.text =
+                        getString(R.string.toolbar_label, it.diceType)
                 }
             }
         }
     }
 
-    override fun onNavigationItemSelected(selectedItem: MenuItem): Boolean {
-        val navigationMenu = binding.navView
-        val fragmentContainer = binding.fragmentContainerView.id
-        val diceFragment = R.id.diceFragment
-        val listOfDiceFragment = R.id.rolledDicesListFragment
-        when(selectedItem.itemId) {
-            R.id.D4DiceMenuItem -> {
-                navigateToFragment(fragmentContainer, diceFragment)
-                viewModel.triggerD4()
-            }
-
-            R.id.D6DiceMenuItem -> {
-                navigateToFragment(fragmentContainer, diceFragment)
-                viewModel.triggerD6()
-            }
-
-            R.id.D8DiceMenuItem -> {
-                navigateToFragment(fragmentContainer, diceFragment)
-                viewModel.triggerD8()
-            }
-
-            R.id.D10DiceMenuItem -> {
-                navigateToFragment(fragmentContainer, diceFragment)
-                viewModel.triggerD10()
-            }
+    private fun setListeners() {
+        binding.menuButton.setOnClickListener { toggleNavMenu() }
+        binding.navView.setNavigationItemSelectedListener(this)
+    }
 
-            R.id.D12DiceMenuItem -> {
-                navigateToFragment(fragmentContainer, diceFragment)
-                viewModel.triggerD12()
+    private fun toggleNavMenu() {
+        with(binding) {
+            when (navView.visibility) {
+                View.GONE -> navView.visibility = View.VISIBLE
+                View.VISIBLE -> navView.visibility = View.GONE
+                else -> navView.visibility = View.GONE
             }
+        }
+    }
```

Comment:
I left this mostly untouched. But it can be improved. I think you should do that. The problem to solve is that when should be exhaustive, and you accomplish that by using `else`. But this is kind of smelly, you aren't be explicit.